src/web_voice_processor.ts

```diff
@@ -92,37 +81,85 @@ export class WebVoiceProcessor {
    * `.postMessage({command: 'process', inputFrame: inputFrame})`.
    * @param engine The engine to unsubscribe.
    */
-  public subscribe(engine: PvEngine): void {
+  public static async subscribe(engine: PvEngine): Promise<void> {
     if (engine.worker) {
       if (engine.worker.postMessage && typeof engine.worker.postMessage === 'function') {
-        this._engines.add(engine);
+        this.instance()._engines.add(engine);
       } else {
         throw new Error("Engine must have a 'onmessage' handler.");
       }
     } else {
       if (engine.postMessage && typeof engine.postMessage === 'function') {
-        this._engines.add(engine);
+        this.instance()._engines.add(engine);
       } else if (engine.onmessage && typeof engine.onmessage === 'function') {
-        this._engines.add(engine);
+        this.instance()._engines.add(engine);
       } else {
         throw new Error("Engine must have a 'onmessage' handler.");
       }
     }
+
+    if (this.instance()._engines.size > 0 && this.instance()._state !== WvpState.STARTED) {
+      await this.instance().start();
+    }
   }
 
   /**
    * Unsubscribe an engine.
    * @param engine The engine to unsubscribe.
    */
-  public unsubscribe(engine: PvEngine): void {
-    this._engines.delete(engine);
+  public static async unsubscribe(engine: PvEngine): Promise<void> {
```

Comment:
Same thing for unsubscribe. Probably be most efficient to allow for an array of engines as well.